src/utils/thread_helper.cpp

```diff
@@ -92,8 +92,24 @@ void ThreadHelper::StateChangedCallback(otChangedFlags aFlags)
         {
             if (mAttachHandler != nullptr)
             {
-                mAttachHandler(OT_ERROR_NONE);
-                mAttachHandler = nullptr;
+                if (mAttachPendingDatasetTlvs.mLength == 0)
+                {
+                    mAttachHandler(OT_ERROR_NONE);
+                    mAttachHandler = nullptr;
+                }
+                else
+                {
+                    otOperationalDataset emptyDataset = {};
+                    otError              error =
+                        otDatasetSendMgmtPendingSet(mInstance, &emptyDataset, mAttachPendingDatasetTlvs.mTlvs,
+                                                    mAttachPendingDatasetTlvs.mLength, MgmtSetResponseHandler, this);
+                    if (error != OT_ERROR_NONE)
+                    {
+                        mAttachHandler(error);
```

Comment:
Maybe little chance, however what's the expecting operations if `otDatasetSendMgmtPendingSet` fails due to either NO_BUF or BUSY error? Do we want the upper layer to retry?

Reply:
I think in this case, it will return the error, and the caller of this API should be responsible to retry.